UsefulProteomicsDatabases/ProteinDbRetriever.cs

```diff
@@ -128,17 +128,38 @@ public static string DownloadAvailableUniProtProteomes(string filepath)
             {
                 Dictionary<string, string> dictionaryOfAvailableProteomes = new Dictionary<string, string>();
                 List<string> idNameList = new List<string>();
-                if (Path.GetExtension(completePathToAvailableUniProtProteomes) == ".gz")
+
+                string fileExtension = Path.GetExtension(completePathToAvailableUniProtProteomes);
+
+                switch (fileExtension)
                 {
-                    idNameList = ReadAllZippedLines(completePathToAvailableUniProtProteomes).ToList();
-                    foreach (string item in idNameList)
-                    {
-                        var lineValuesArray = item.Split("\t");
-                        dictionaryOfAvailableProteomes.Add(lineValuesArray[0], lineValuesArray[1]);
-                    }
-                    return dictionaryOfAvailableProteomes;
-                }
-                return null;
+                    case ".gz":
+                        idNameList = ReadAllGZippedLines(completePathToAvailableUniProtProteomes).ToList();
+                        foreach (string item in idNameList)
+                        {
+                            var lineValuesArray = item.Split("\t");
+                            dictionaryOfAvailableProteomes.Add(lineValuesArray[0], lineValuesArray[1]);
+                        }
```

Comment:
I noticed this set of lines is copy-pasted many times. I think it may be better if a method is instead added to the ProteinDbRetriever class.